### ISSUE TITLE
Support a proper dry run

### ISSUE THREAD

**Issue by stevelacey** [prantlf/jsonlint#5]
Jsonlint will exit 0 even when there are still things that `--in-place` would fix, which means to do a full lint that will fail a CI pipeline, you have to run `--in-place` and assess whether there were any resulting changes e.g. something like:

```bash
jsonlint resources/lang --quiet --in-place --indent="  " --no-duplicate-keys --sort-keys \
 && git --no-pager diff resources/lang \ # to show the diff
 && git diff --quiet --stat resources/lang  # to exit 1
```

Would be great if there was a way to make key sorting and incorrect indentation error out.

**Comment by prantlf** [comment]
I see. You would like to check that there is no difference between the original content and  the pretty-printed output from `jsonlint`. A clever idea with the `git diff`!

How about a `compare` switch?

    jsonlint resources/lang --log-files --compare \
      --indent="  " --no-duplicate-keys --sort-keys

Print name of the each processed file, compare that a pretty-printed version is the same as the original, specify indentation, forbid duplicate keys and request key sorting.

**Comment by stevelacey** [comment]
I don't work on the project I used jsonlint on anymore, maybe `--compare` makes sense here, but ` --check` / `--dry-run` is what every other linter I use names this.

```
eslint (check is the default behavior, `--fix` fixes)
stylelint (check is the default behavior, `--fix` fixes)
black --check
isort --check
python manage.py makemigrations --check --dry-run
```

Specifically, I'd just want to make it easy to fail a CI pipeline when jsonlint fixes are due, I don't care much about the output, but sure, there should be output, a list of files that would be changed, and/or a diff.

**Comment by prantlf** [comment]
Thanks for the quick reply and the ideas! For the purpose of the formatting check I like the `check` option more than `dry-run`, because it suggests that it will check and might fail. The `dry-run` suggests only that it would do the work without saving changes.

I added a command line argument `--check`, which is supposed to be used instead of `--in-place`. The validation and formatting will take place, but the output will not be written out. It will be compared to the input and if it differs, it will fail the processing and dump the unified diff between the original and formatted content.

**Comment by prantlf** [comment]
I added one more option - `--diff` - to print the diffs instead of the content. It can be used as a dry-run without checking and failing.
